sdks/Cordova.md

```diff
@@ -0,0 +1,136 @@
+
+## Architecture for solution 1
+
+### Config parsing 
+
+Cordova config plugin not needed due to trivial implementation that can be done in pure JavaScript.
+Networking/Logging should be done on JavaScript side as this is common aproach for cordova.
+Due to trivial implementation no need for core module like in Android and IOS.
+
+### Individual SDK's
+
+Using existing cordova plugins or creating new cordova plugins for services.
+Following example is using auth module, but behaviour should be the same for the most of the plugins.
+Example:
+
+```
+// FIXME We need option without using browserify/webpack
+var config = require('mobile-config.json')
+var ConfigLib = require('aerogear-services')
+
+var keycloakConfig = ConfigLib(config).getKeyCloakConfig()
+
+cordova.auth.init(keycloak)
+
+// Using 
+cordova.auth.login()
+// Allow to retrieve token in javascript.
+cordova.auth.getToken()
+```
+
+This code relects how cordova based solutions work in general.
+
+## Investigation process
+
+All investigations and different variations of architecture will be available in following repository:
+https://github.com/wtrocki/CordovaSandbox
+
+### POC for Core
+
+Two options were investigated
+1. Wrapping core into plugin. 
+1. Writing pure JS helper.
+
+Wrapping core into plugin was proven difficult and innefective:
+- JavaScript implementation is trivial
+- Droping configuration into native projects may be bad user experience for cordova developers.
+- Core android IOS providing helpers that are available on cordova/javascript
+- Only parser needed to be linked back to javascript
+- The way Android instantiates service will not be compatibile with Cordova 
+- Wrapping core into cordova plugin will put some limitations on native implementation we want to avoid.
+
+JavaScript configuration parser/processor was implemented.
+This was proven to be the most efficient way to resolve configuration parsing problem.
+
+### Interactions between services (binding)
+
+TODO Link github example
+
+### Investigating individual services
+
+### Auth
+
+TODO Document
+
+Source: https://github.com/wtrocki/CordovaSandbox/tree/master/aerogear-auth-plugin
+
+### Push
+
+TODO Document
+Evaluating existing plugin: https://github.com/aerogear/aerogear-cordova-push
+
+### Metrics
+TODO Document
+
+Suggestion/Spike: metrics done in pure JavaScript/Typescript.
+Challenge: How to push from cordova plugins. Can we pull for cordova?
+
+# Possible limitations
+
+Solution will be able to reuse existing cordova plugins like push, 
+however to effective manage and build cordova plugins, 
+both IOS and Android AeroGear Services SDK should not contain any SDK specific code.
+Instead it should just link to the library that can be then also wrapped to the Cordova plugin on different repository.
+
+I identified a couple solutions to extract existing code from SDK, 
```

Comment:
As per comment above, lets not get bogged down in the various service SDKs or how existing SDKs (like push) can be used.